src/marks/auto.js

```diff
@@ -0,0 +1,338 @@
+        if (new Set(color).size > color?.length >> 1) {
+          // TODO isHighCardinality(color)
+          // TODO only if z not set by user
+          z = null;
+        }
+        break;
+      case "area":
+        // TODO: throw some errors; I'm not sure we ever want Plot.area; how does it work with ordinal?
+        mark =
+          x && y
+            ? isContinuous(x) && isMonotonic(x)
+              ? areaY
+              : isContinuous(y) && isMonotonic(y)
+              ? areaX
+              : area
+            : x
+            ? areaX
+            : areaY; // 1d area by index
+        fill = color;
+        fillReduce = colorReduce;
+        break;
+      case "rule":
+        mark = x ? ruleX : ruleY;
+        stroke = color;
+        strokeReduce = colorReduce;
+        break;
+      case "bar":
+        mark =
+          yReduce != null
+            ? isOrdinal(x)
+              ? barY
+              : rectY
+            : xReduce != null
+            ? isOrdinal(y)
+              ? barX
+              : rectX
+            : colorReduce != null
+            ? isOrdinal(x) && isOrdinal(y)
+              ? cell
+              : isOrdinal(x)
+              ? barY
+              : isOrdinal(y)
+              ? barX
+              : rect
+            : isOrdinal(x) && isOrdinal(y)
+            ? cell
+            : isOrdinal(y)
+            ? barX
+            : barY;
+        fill = color;
+        fillReduce = colorReduce;
+        break;
+      default:
+        throw new Error(`invalid mark: ${mark}`);
+    }
+  }
+
+  // Determine the mark options.
+  let options = {x, y, fill, stroke, z, r: size, fx, fy};
+  let transformOptions = {
+    fill: fillReduce,
+    stroke: strokeReduce,
+    z: zReduce,
+    r: sizeReduce
+  };
+  if (xReduce != null && yReduce != null) {
+    throw new Error(`cannot reduce both x and y`); // for now at least
+  } else if (yReduce != null) {
+    options = isOrdinal(x)
+      ? groupX({y: yReduce, ...transformOptions}, options)
+      : binX({y: yReduce, ...transformOptions}, options);
+  } else if (xReduce != null) {
+    options = isOrdinal(y)
+      ? groupY({x: xReduce, ...transformOptions}, options)
+      : binY({x: xReduce, ...transformOptions}, options);
+  } else if (colorReduce != null || sizeReduce != null) {
+    if (isOrdinal(x) && isOrdinal(y)) {
+      options = group(transformOptions, options);
+    } else if (!isOrdinal(x) && !isOrdinal(y)) {
+      options = bin(transformOptions, options);
+    } else if (isOrdinal(x) && !isOrdinal(y)) {
+      options = binY(transformOptions, options); // bin y, group x
+    } else if (!isOrdinal(x) && isOrdinal(y)) {
+      options = binX(transformOptions, options); // bin x, group y
+    }
+  }
+
+  return marks(
+    // x: { zero: xZero },
+    // y: { zero: yZero },
+    // color: colorValue == null && colorReduce == null ? null : { legend: true },
+    // marks: [
+    fx || fy ? frame({stroke: "#eee"}) : null,
```

Comment:
I don't have a strong opinion on whether we have a frame by default. I think especially with faceted dots I find it hard to see the boundaries; for someone unfamiliar with faceting, I think the frame makes it more clear what it's doing. With bars it's less of an issue. (But I'll definitely at least update to match grid.)

<img width="758" alt="image" src="https://user-images.githubusercontent.com/841829/215566465-0a24f15d-b1e6-40a5-830b-9ce8863f5edb.png">

Reply:
I find myself adding a frame almost every time I use facets and quantitative x or y, because there's a kind of "visual continuity" that needs to be broken. (When x and y are ordinal it's ok not to have it.)

Reply:
Okay, let’s do it. Seems like maybe we should also file a feature request to have Plot do this by default when faceting though!